src/js/videojs-dash.js

```diff
@@ -89,6 +91,12 @@ class Html5DashJS {
     this.mediaPlayer_.setProtectionData(this.keySystemOptions_);
     this.mediaPlayer_.attachSource(manifestSource);
 
+    // Setup text tracks
+    setupTextTracks.call(null, this.player, tech);
```

Comment:
This makes sense. If the user can fetch a source and attach it, this might become a sync call and the events might get missed. I've moved the calls to be before `attachSource()`.